File: gestor-fe-backend/gestor-fe-core-service/src/main/java/com/gestor_fe/core/service/PrestadorService.java
package com.gestor_fe.core.service;

import java.util.Optional;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

import com.gestor_fe.core.entity.Documento;
import com.gestor_fe.core.entity.Prestador;

public interface PrestadorService {

    // --- Operaciones del Prestador (CRUD / Maestro) ---
    Prestador crearOActualizarPrestador(Prestador prestador);
    Optional<Prestador> obtenerPorNit(String nit);
    Optional<Prestador> obtenerPorId(Long id);
    Page<Prestador> listarPrestadores(Pageable pageable);

    // --- Operaciones de Soportes / Documentos del Prestador ---
    Documento cargarSoporte(String nitPrestador, Long tipoId, Long extensionId, MultipartFile archivo);
    Page<Documento> listarSoportes(Long prestadorId, Pageable pageable);
    void eliminarSoporte(Long documentoId);
}

File: gestor-fe-frontend/src/app/services/prestador.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CommonService } from './common.service';
import { Documento } from '../models/documento';
import { BECORE } from '../config/app';
import { Prestador } from '../models/prestador';

@Injectable({
  providedIn: 'root'
})
export class PrestadorService extends CommonService<Prestador> {

  protected override endPointBase = BECORE + '/api/v1/prestadores';

  constructor(http: HttpClient) {
    super(http);
  }

  public obtenerPorNit(nit: string): Observable<Prestador> {
    return this.http.get<Prestador>(`${this.endPointBase}/nit/${nit}`);
  }

  public cargarSoporte(nitPrestador: string, tipoId: number, extensionId: number, archivo: File): Observable<Documento> {
    const formData = new FormData();
    formData.append('nitPrestador', nitPrestador);
    formData.append('tipoId', tipoId.toString());
    formData.append('extensionId', extensionId.toString());
    formData.append('archivo', archivo);

    return this.http.post<Documento>(`${this.endPointBase}/soportes/cargar`, formData);
  }

  public listarSoportes(prestadorId: number, page: number = 0, size: number = 20): Observable<any> {
    const params = new HttpParams()
      .set('page', page.toString())
      .set('size', size.toString());

    return this.http.get<any>(`${this.endPointBase}/${prestadorId}/soportes`, { params });
  }

  public eliminarSoporte(documentoId: number): Observable<void> {
    return this.http.delete<void>(`${this.endPointBase}/soportes/${documentoId}`);
  }
}
